mlrun/api/api/endpoints/model_endpoints.py

```diff
@@ -62,15 +72,62 @@ def create_or_patch(
             f"Mismatch between endpoint_id {endpoint_id} and ModelEndpoint.metadata.uid {model_endpoint.metadata.uid}."
             f"\nMake sure the supplied function_uri, and model are configured as intended"
         )
-    # Since the endpoint records are created automatically, at point of serving function deployment, we need to use
-    # V3IO_ACCESS_KEY here
-    mlrun.api.crud.ModelEndpoints().create_or_patch(
+
+    mlrun.api.crud.ModelEndpoints().create_model_endpoint(
         db_session=db_session,
         access_key=os.environ.get("V3IO_ACCESS_KEY"),
         model_endpoint=model_endpoint,
+    )
```

Comment:
As this is supposed to be run through by the API, I think querying the environment variable one time is enough to decide whether the env exists or not, and just save it as a config attribute. That way you won't have to query or pass that to every function, but rather each function that needs the param will just access the config.

Take a look at `resolve_chief_api_url` in config.py for reference.